src/db.ts

```diff
@@ -1,27 +1,204 @@
 import mongoose from "mongoose";
 import * as dotenv from 'dotenv';
+import { RoomSchema, QuizSchema, QuestionSchema } from './database/schema';
+import { getModelForClass } from "@typegoose/typegoose";
 
-// get env variables
 dotenv.config();
 
 if (process.env.MONGO_URI === undefined) {
-    console.error('env variable MONG_URI is not set in .env file');
-    // exit program
+    console.error('env variable MONGO_URI is not set in .env file');
     process.exit(1);
 }
 
 const uri = process.env.MONGO_URI;
 
-async function run() {
-    try {
+class dbHandler {
+    private static instance: dbHandler | null = null;
+    
+    private constructor() {
+        this.connect().catch((error) => {
+            console.error("Database connection error:", error);
+            process.exit(1);
+        });
+    }
+
+    public static getInstance(): dbHandler {
+        if (!dbHandler.instance) {
+            dbHandler.instance = new dbHandler();
+        }
+        return dbHandler.instance;
+    }
+
+    private async connect(): Promise<void> {
         await mongoose.connect(uri);
-        await mongoose.connection.db.admin().command({ ping: 1 });
-        console.log("successfully connected to DB");
-    } finally {
+    }
+
+    private async disconnect(): Promise<void> {
         await mongoose.disconnect();
     }
+
+    public async createRoom(roomData: RoomSchema): Promise<void> {
+        try {
+            const roomModel = getModelForClass(RoomSchema);
+            const room = new roomModel({ HostWsId: roomData.HostWsId, PlayerIds: roomData.PlayerIds, RoomState: roomData.RoomState});
+            await room.save();
+            console.log(`New room created with id (handler): ${room._id}`);
+        } catch (error) {
+            console.error("Room create error (handler):", error);
+        }
+    }
+
+    public async updateRoomRecord(roomId: string, updateData: {
+        RoomState?: string,
+        PlayerIds?: Array<string>,
+        HostWsId?: string
+    }): Promise<void> {
+        const roomModel = getModelForClass(RoomSchema);  
+        try {
+            const res = await roomModel.updateOne({ _id: roomId }, updateData);
+            console.log(`Room updated with id (handler): ${roomId}`);
+        } catch (error) {
+            console.error("Room update error (handler):", error);
+        }
+    }
+
+    public async createQuiz(quizData: QuizSchema): Promise<void> {
+        try {
+            const quizModel = getModelForClass(QuizSchema);
+            const quiz = new quizModel({ Questions: quizData.Questions });
+            await quiz.save();
+            console.log(`New quiz created with id (handler): ${quiz._id}`);
+        } catch (error) {
+            console.error("Quiz create error (handler):", error);
+        }
+    }
+
+    public async updateQuizList(quizId: string, updateData: {
```

Comment:
what does `updateQuizList` do? Does it update a quiz? or update a list field in the quiz model? The name of the function matters a lot. Please rename this function to match its functionality appropriately

Reply:
any follow up? @geneh61

Reply:
it updates the list of question ids in the quiz model, would a good name be updateQuizIdList or updateQuestionList or something else? I'm having trouble coming up with a concise name